<center>
    <img src="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DS0105EN-SkillsNetwork/labs/Module2/images/SN_web_lightmode.png" width="300" alt="cognitiveclass.ai logo">
</center>


#### Add your code below following the instructions given in the course


### Topic: Analyzing the Iris Dataset with Python

In this notebook, I will analyze the famous Iris dataset using Python. The Iris dataset is a classic dataset in machine learning and consists of measurements of the sepal length, sepal width, petal length, and petal width of three species of Iris flowers.

Here's an outline of how you could structure your notebook:

# My Jupyter Notebook on IBM Watson Studio

Name: [Ekofiongo Eale]<br>
Data Science Enthusiast

I am interested in data science because it allows me to make sense of complex data and gain insights that can be used to make better decisions.

### Code Explanation

The code in this cell will load the Iris dataset and display the first 5 rows.

In [7]:
import pandas as pd

# Load the Iris dataset from the scikit-learn library
from sklearn.datasets import load_iris
iris = load_iris()

# Convert the dataset to a pandas DataFrame and display the first 5 rows
df = pd.DataFrame(data=iris.data, columns=iris.feature_names)
df.head()

sepal length (cm)  sepal width (cm)  petal length (cm)  petal width (cm)
0                5.1               3.5                1.4               0.2
1                4.9               3.0                1.4               0.2
2                4.7               3.2                1.3               0.2
3                4.6               3.1                1.5               0.2
4                5.0               3.6                1.4               0.2

### Data Description

The Iris dataset contains measurements of the sepal length, sepal width, petal length, and petal width of three species of Iris flowers: setosa, versicolor, and virginica.

The dataset contains 150 samples, with 50 samples from each species. The goal of this analysis is to explore the relationships between the different features and the species.

Here's a bulleted list of some of the questions we'll answer in this notebook:

What is the distribution of each feature in the dataset?
Is there a correlation between the features?
Can we use the features to predict the species of an Iris flower?

Finally, here's a hyperlink to the scikit-learn documentation for the Iris dataset: https://scikit-learn.org/stable/auto_examples/datasets/plot_iris_dataset.html

In [8]:
import pandas as pd
from sklearn.datasets import load_iris
from sklearn.tree import DecisionTreeClassifier
from sklearn.model_selection import train_test_split
from sklearn.metrics import accuracy_score

# Load the Iris dataset from the scikit-learn library
iris = load_iris()

# Convert the dataset to a pandas DataFrame
df = pd.DataFrame(data=iris.data, columns=iris.feature_names)
df['species'] = iris.target

# Split the data into training and testing sets
X_train, X_test, y_train, y_test = train_test_split(df[iris.feature_names], df['species'], random_state=0)

# Create a decision tree classifier and fit it to the training data
clf = DecisionTreeClassifier(max_depth=2, random_state=0)
clf.fit(X_train, y_train)

# Make predictions on the testing data
y_pred = clf.predict(X_test)

# Calculate the accuracy of the model
accuracy = accuracy_score(y_test, y_pred)

# Print the accuracy
print(f"Accuracy: {accuracy:.2f}")


Accuracy: 0.89


This code loads the Iris dataset, converts it to a pandas DataFrame, splits it into training and testing sets, creates a decision tree classifier with a maximum depth of 2, fits the classifier to the training data, makes predictions on the testing data, and calculates the accuracy of the model. The output shows that the model has an accuracy of 0.89, which is quite good.

Note that this is just an example, and there are many other machine learning models that could be used with the Iris dataset.